OsmAnd/src/net/osmand/plus/helpers/LockHelper.java

```diff
@@ -0,0 +1,150 @@
+package net.osmand.plus.helpers;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Handler;
+import android.os.PowerManager;
+
+import net.osmand.AndroidUtils;
+import net.osmand.PlatformUtil;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.routing.VoiceRouter;
+
+public class LockHelper implements SensorEventListener {
+	private static final org.apache.commons.logging.Log LOG = PlatformUtil.getLog(LockHelper.class);
+
+	private static final int SENSOR_SENSITIVITY = 4;
+
+	private final SensorManager mSensorManager;
+	private final Sensor mProximity;
+	private PowerManager.WakeLock wakeLock = null;
+	private Handler uiHandler;
+	private OsmandApplication app;
+	private LockRunnable lockRunnable;
+	private LockUIAdapter lockUIAdapter;
+
+	private VoiceRouter.VoiceMessageListener voiceMessageListener;
+
+	public interface LockUIAdapter {
+
+		void lock();
+
+		void unlock();
+	}
+
+	private class LockRunnable implements Runnable {
+		@Override
+		public void run() {
+			lock();
+		}
+	}
+
+	public LockHelper(final OsmandApplication app) {
+		this.app = app;
+		uiHandler = new Handler();
+		lockRunnable = new LockRunnable();
+
+		mSensorManager = (SensorManager) app.getSystemService(Context.SENSOR_SERVICE);
+		mProximity = mSensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY);
+
+		voiceMessageListener = new VoiceRouter.VoiceMessageListener() {
+			@Override
+			public void onVoiceMessage() {
+				unlockEvent();
+			}
+		};
+
+		setVoiceRouterListener(true);
```

Comment:
Why it is not the same - https://github.com/osmandapp/Osmand/commit/f9cd73d4f02b45c429a94e5444124cc6f3463111#diff-33da24854a630e0918b813683c839fbbL79